packages/mui-material/src/styles/overrides.ts

```diff
@@ -127,13 +127,10 @@ export type OverridesStyleRules<
     // Record<string, unknown> is for other props that the slot receive internally
     // Documenting all ownerStates could be a huge work, let's wait until we have a real needs from developers.
     (ComponentName extends keyof ComponentsPropsList
-      ? ComponentsPropsList[ComponentName] &
-          Record<string, unknown> & {
-            ownerState: ComponentsPropsList[ComponentName] & Record<string, unknown>;
-          }
-      : {}) & {
+      ? ComponentsPropsList[ComponentName] & Record<string, unknown>
```

Comment:
Moved `ownerState` definition inside `CSSObjectWithVariants` to make it closer to `ComponentsVariants` type